src/main/java/org/jenkinsci/plugins/kubernetes/auth/KubeConfigBuilder.java

```diff
@@ -0,0 +1,36 @@
+package org.jenkinsci.plugins.kubernetes.auth;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.fabric8.kubernetes.api.model.AuthInfoBuilder;
+import io.fabric8.kubernetes.api.model.Cluster;
+import io.fabric8.kubernetes.client.internal.SerializationUtils;
+import org.jenkinsci.plugins.kubernetes.credentials.Utils;
+
+public class KubeConfigBuilder {
```

Comment:
```suggestion
abstract class KubeConfigBuilder implements KubernetesAuth {
```

Also, I would suggest to change the class name to something like `AbstractKubernetesAuth`. Usually the suffix `Builder` indicates the class follows the [Builder](https://dzone.com/articles/design-patterns-the-builder-pattern) pattern.